### ISSUE TITLE
onComplete acts when rendering has not finished yet

### ISSUE THREAD

**Issue by guillemdc** [swagger-api/swagger-ui#3475]
- swagger-ui version: v3.0.19
- a swagger file reproducing the issue: any file

We have a table in which we list all the operations from all the APIs we have live at the moment. In SwaggerUI 2 we used to indicate in the hash the operation from the API we needed so, when clicking the link, you'd be able to open the required API and, once rendered, swagger UI would scroll to the required operation and maximize it.

Now, trying to migrate to SwaggerUI 3, we use the following code:

```
function goToHash(){
	if(window.location.hash) {
		$(window.location.hash).children().first().click()
		$('#wrapper').animate({
			scrollTop: $(window.location.hash).offset().top
		}, 1000);
	}
}
```

```
const ui = SwaggerUIBundle({
	url: '/api-docs/<?php echo $this->data['id']; ?>',
	dom_id: "#wrapper",
	presets: [
		 SwaggerUIBundle.presets.apis
	],
	plugins: [
	  SwaggerUIBundle.plugins.DownloadUrl
	],
	onComplete: goToHash()
})
```

However, goToHash runs before SwaggerUI has rendered and hence, the div with the id we are looking for is not in the dom yet and fails.

**Comment by shockey** [comment]
Hi @guillemdc!

`onComplete` is not yet implemented in v3. Your function is probably firing when SwaggerUI is called, because in your example you're invoking it (`goToHash()`) instead of passing the function definition into onComplete (i.e. `onComplete: goToHash`).

As for deep linking via a hash: this was recently implemented in v3, please see https://github.com/swagger-api/swagger-ui/issues/2884 and #3388 for details. The functionality should be backwards-compatible with the v2 links, but please note that it is _not_ enabled by default as it was in v2, you need to enable it via a config option. Our reasons for that are laid out in the PR for the feature, if you're interested.

Let me know if this helps. Thanks for reaching out!

**Comment by guillemdc** [comment]
Hi @shockey 

I had it wrapped into a function but, as it didn't get triggered I unwrapped it thinking that maybe it was implemented somehow different. As it started working I thought it was implemented, but I was wrong! So lame I did not realize that if I unwrapped it the function would be called no matter what.

I've enabled the deep links and now everything works flawlessly. As you say, it is backwards compatible and our old links are still working 😄 

Thanks a lot for the quick and great support!
